### PR TITLE
Scal and swap 64-bit interface

### PR DESCRIPTION
**Pull request by daineAMD**

### CODE REVIEW

**Review by TorreZuk — APPROVED** [review]
LGTM, hopefully with the rebase you'll get all passing now

**Comment by TorreZuk** [comment]
@daineAMD I think the cuda is due to Eiden changing the docker again build to #13279 , as my fix PR all passed cuda before I merged :-)

**Comment by TorreZuk** [comment]
http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin-cuda%2FhipBLAS/detail/PR-759/2/pipeline

File: .jenkins/precheckin-cuda.groovy [hunk 1]
```diff
@@ -89,6 +89,6 @@ ci: {
             properties(auxiliary.addCommonProperties(property))
     }
 
-    String hostBuildCommand = './install.sh -c --compiler=g++ --cuda'
+    String hostBuildCommand = 'HIP_PLATFORM=nvidia ./install.sh -c --compiler=g++'
```
Comment (TorreZuk) [inline .jenkins/precheckin-cuda.groovy]: Okay so this is wrong @daineAMD.  Nothing in install.sh reads that env variable to set build configuration.  
I would suggest this needs to be added in separate PR.  Also the move from install.sh code duplication of rmake.py should be addressed soon.  I was trying to explain whether CI failures which seem to be using wrong reference library were due to CI pipeline bug or stale cache but this should be reverted first to see if it passes.  Then I can make fix PR or you can, we can discuss.  Same goes for moving build code from install into rmake I can look if you want.

Comment (daineAMD) [reply .jenkins/precheckin-cuda.groovy]: Yep I'll remove that now. I do have a ticket for fixing install.sh/rmake.py duplication, may or may not get to it this release.

**Comment by daineAMD** [comment]
Windows CI failed but passed previous build so fine to merge I think.

**Pull request merged**